File: app/src/main/java/com/m7mdabaza/fooddelivery/onBoarding/activities/OnBoardingActivity.java
package com.m7mdabaza.fooddelivery.onBoarding.activities;

import androidx.appcompat.app.AppCompatActivity;
import androidx.viewpager.widget.ViewPager;

import android.os.Bundle;
import android.content.Intent;
import android.content.SharedPreferences;
import android.content.Context;
import android.widget.TextView;

import com.google.android.material.tabs.TabLayout;
import com.m7mdabaza.fooddelivery.R;
import com.m7mdabaza.fooddelivery.onBoarding.FragmentAdapter;
import com.m7mdabaza.fooddelivery.ui.activities.HomeActivity;

public class OnBoardingActivity extends AppCompatActivity {

    String prevStarted = "prevStarted";

    @Override
    protected void onResume() {
        super.onResume();
        SharedPreferences sharedpreferences = getSharedPreferences(getString(R.string.app_name), Context.MODE_PRIVATE);
        if (!sharedpreferences.getBoolean(prevStarted, false)) {
            SharedPreferences.Editor editor = sharedpreferences.edit();
            editor.putBoolean(prevStarted, Boolean.TRUE);
            editor.apply();
        } else {
            //finish();
            Intent intent = new Intent(OnBoardingActivity.this, HomeActivity.class);
            startActivity(intent);
        }
    }

    TextView getStarted;
    ViewPager viewPager;
    TabLayout tabLayout;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_on_boarding);

        viewPager = findViewById(R.id.viewpager);
        getStarted = findViewById(R.id.getStarted);
        tabLayout = findViewById(R.id.tabDots);

        FragmentAdapter fragmentAdapter = new FragmentAdapter(getSupportFragmentManager());
        viewPager.setAdapter(fragmentAdapter);

        tabLayout.setupWithViewPager(viewPager, true);

        getStarted.setOnClickListener(view -> {
            Intent intent = new Intent(OnBoardingActivity.this, HomeActivity.class);
            startActivity(intent);
        });

    }
}

File: app/src/main/java/com/m7mdabaza/fooddelivery/onBoarding/activities/OnBoardingActivity.kt
package com.m7mdabaza.fooddelivery.onBoarding.activities

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.m7mdabaza.fooddelivery.R
import com.m7mdabaza.fooddelivery.onBoarding.adapters.FragmentAdapter
import com.m7mdabaza.fooddelivery.ui.activities.HomeActivity
import kotlinx.android.synthetic.main.activity_on_boarding.*
import androidx.viewpager2.widget.ViewPager2

class OnBoardingActivity : AppCompatActivity() {

    private lateinit var viewPager2: ViewPager2

    private var prevStarted = "prevStarted"

    override fun onResume() {
        super.onResume()
        val sharedpreferences = getSharedPreferences(getString(R.string.app_name), MODE_PRIVATE)
        if (!sharedpreferences.getBoolean(prevStarted, false)) {
            val editor = sharedpreferences.edit()
            editor.putBoolean(prevStarted, true)
            editor.apply()
        } else {
            val intent = Intent(this, HomeActivity::class.java)
            startActivity(intent)
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_on_boarding)

        val fragmentAdapter = FragmentAdapter(activity = this, 3)
        viewPager2 = findViewById(R.id.viewpager2)
        viewPager2.adapter = fragmentAdapter

        tabDots.setViewPager(viewPager2)
        //Optional
        fragmentAdapter.registerAdapterDataObserver(tabDots.adapterDataObserver)

        getStarted.setOnClickListener {
            val intent = Intent(this, HomeActivity::class.java)
            startActivity(intent)
        }

    }

}